### PR TITLE
GPU markers support

### LINKED ISSUE #557: Annotate passes/draw calls
OpenGL ES may support `KHR_Debug` and/or `EXT_debug_marker`, as mentioned on the apitrace usage page: https://github.com/apitrace/apitrace/blob/master/docs/USAGE.markdown#opengl-annotations
By adding the annotations we could greatly enhance the debugging experience.

### PR DESCRIPTION
**Pull request by kvark**
Fixes #557

---
This change is [Reviewable](https://reviewable.io/reviews/servo/webrender/604)

### CODE REVIEW

File: webrender/src/device.rs [hunk 1]
```diff
@@ -639,6 +639,36 @@ impl<T> GpuProfiler<T> {
     }
 }
 
+pub struct GpuMarker(());
```
Comment (glennw) [inline webrender/src/device.rs]: Can this just be pub struct GpuMarker; ?

File: webrender/src/renderer.rs [hunk 2]
```diff
@@ -1076,6 +1077,7 @@ impl Renderer {
                    target_size: &DeviceSize,
                    cache_texture: Option<TextureId>,
                    should_clear: bool) {
+        let _ = GpuMarker::new("target draw");
```
Comment (glennw) [inline webrender/src/renderer.rs]: Instead of these hard coded strings, could we use the label field that exists in the GpuProfileTag structure?

Comment (kvark) [inline webrender/src/renderer.rs]: Oh, I missed the fact that those have names. Going to have the `add_marker` returning a `GpuMarker` object then, unless you think of a better way ;)

File: webrender/src/device.rs (hunk 1 above)
Comment (kvark) [inline webrender/src/device.rs]: No, since it's unsafe to create an instance of it without calling `GpuMarker::new()`

**Comment by kvark** [comment]
Updated now. How does this look?

**Pull request merged**